### PR TITLE
Update to Groovy 2.5.0 - 2nd attempt

### PR DESCRIPTION
**Pull request by aaloise**
@vietj this time it is compiling, at least for me. Please, can you check if this PR is enough?

### CODE REVIEW

File: vertx-lang-groovy-codegen/pom.xml [hunk 1]
```diff
@@ -22,8 +22,13 @@
     </dependency>
     <dependency>
       <groupId>org.codehaus.groovy</groupId>
-      <artifactId>groovy-all</artifactId>
-      <version>2.4.15</version>
+      <artifactId>groovy</artifactId>
+      <version>2.5.0</version>
+    </dependency>    
+    <dependency>
+      <groupId>org.codehaus.groovy</groupId>
```
Comment (vietj) [inline vertx-lang-groovy-codegen/pom.xml]: why is groovy-console required ?

File: vertx-lang-groovy-codegen/pom.xml [hunk 2]
```diff
@@ -80,12 +85,6 @@
       <scope>test</scope>
     </dependency>
     <!-- Used by failsafe plugin that excludes the compile scope -->
-    <dependency>
```
Comment (vietj) [inline vertx-lang-groovy-codegen/pom.xml]: I think there is a good reason we have this dependency can you just upgrade it instead of removing it ?

**Comment by aaloise** [comment]
Without groovy-console, VertxTransformation class doesn't compile.

As for the dependency removed, I tried to upgrade. It was the groovy jar with test scope. I removed it because it's presence for some reason bugs the compilation. This happened here with me.

**Comment by vietj** [comment]
can you add a note about VertxTransformation in `pom.xml` ?

can you show what went wrong with the jar upgrade ?

File: vertx-lang-groovy-codegen/pom.xml [hunk 3]
```diff
@@ -22,8 +22,13 @@
...
     <dependency>
       <groupId>org.codehaus.groovy</groupId>
-      <artifactId>groovy-all</artifactId>
-      <version>2.4.15</version>
+      <artifactId>groovy</artifactId>
+      <version>2.5.0</version>
+    </dependency>    
+    <dependency>
+      <groupId>org.codehaus.groovy</groupId>
+      <artifactId>groovy-console</artifactId>
```
Comment (aaloise) [inline vertx-lang-groovy-codegen/pom.xml]: This dependency is needed so VertxTransformation class can compile.

**Comment by aaloise** [comment]
- Note for VertxTransformation class added.
- With the jar upgrade you would see all that compilation errors that happened with #60. Actually even without the upgrade the compile fails, so the only way I found to make the compile to happen was to remove the dependency.

**Comment by aaloise** [comment]
I've done some readings on Maven scopes and maybe these considerations answers why the build was failing with the presence of the dependency I removed with this PR.

In practice, there were two groovy dependencies. One with the compile scope and one with the test scope. Regardless of whether or not you upgrade the test-scoped dependency, compilation gives error (see #60) as if it's presence overwrites the existence of the other dependency, the one with the compile scope.

After many attempts to keep the two dependencies, in which there was always the same compilation error, I tried to remove the test scope dependency and to my surprise the compilation was successful. To be honest I do not know if the test scope dependency will be missed, since the same jar will be provided by compile scope. When reading the Maven documentation on scopes, it seems to me that the compile scope is sufficient.

compile
This is the default scope, used if none is specified. Compile dependencies are available in all classpaths of a project. Furthermore, those dependencies are propagated to dependent projects.

test
This scope indicates that the dependency is not required for normal use of the application, and is only available for the test compilation and execution phases. This scope is not transitive.

Perhaps someone with more knowledge in Maven can clarify if my understanding is correct.

**Pull request closed without merging**